### ISSUE TITLE
Plans to add sync/cleanup behaviour of subjects

### ISSUE THREAD

**Issue by leosilvadev** [ImFlog/schema-registry-plugin#36]
Hi, this is more like a question.
Is it in the plans of the plugin to support some sort of way to sync the subjects? By that I mean: when we deprecate a Subject (like migrated to a new structure), so being forced to remove the deprecated ones manually (or doing a script apart from the plugin).

**Comment by leosilvadev** [comment]
If this is a possible feature for the plugin, let me know. I can also help with the implementation

**Comment by ImFlog** [comment]
Hello, thank you for the interest in this plugin.
I am not sure to understand what you mean by `a way to sync the subjects`.
Do you mean a task that retrieves all the Subjects available in a given registry ?

**Comment by leosilvadev** [comment]
So, by that I mean:
Retrieve all the subjects and compare to the subjects I have configured locally, the new ones will be added/updated as they are now and the ones that exist in the registry but not locally would be deleted.

This is currently one issue on my side since we are migrating a lot of events and we need to remove the old ones at some point.

**Comment by ImFlog** [comment]
Are you planning to run this "sync" task only once ? If so a standard script (python / bash) could do the trick.

If not, I think that if we implement a "removal" task, you could play with the all the download/register/remove and achieve what you want:
* download all schema in a directory
* for each downloaded schemas, build your removal / insert lists by comparing with what you have locally in another directory
* then using the two list, do the insert / delete

WDYT ?

**Comment by leosilvadev** [comment]
I think it could be simpler, at least from what I would need.
We could just check whether a subject is still marked to be registered or not, in case it is not, it will be removed.

Not sure if this makes sense for all the cases, but from my case this is how it works:
- we maintain all the schemas in one repo, so this is the source of truth
- we run our ci/cd pipeline to register the schemas when we have something new, so whatever is mapped in this repo should be inside the registry, nothing more and nothing less, this means also that in case someone register something manually, this would be removed

**Comment by ImFlog** [comment]
I don't think we will add this for multiple reasons:
* The use case is a bit too specific to be generalized in this plugin
* I don't really want to add the DELETE schema call, it should be handled with care and the Gradle plugin (as the maven one) is not the place for this IMHO

Still, I've just noticed that the DownloadSubject in the Maven plugin support regexp, I guess I could update the task to accept it too. Like this you could hack your own task and do what you want (using the steps I described in my previous comment).

**Comment by ImFlog** [comment]
I'm closing this as I stated I think you use case is very specific.
Thank you for your input and do not hesitates to create another issue for anything :)
